core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -215,7 +215,8 @@ private[spark] class PythonRunner(
 
           case e: Exception if context.isInterrupted =>
             logDebug("Exception thrown after task interruption", e)
-            throw new TaskKilledException
+            context.killTaskIfInterrupted()
+            null  // not reached
```

Comment:
nit: It would be good if we could directly throw the exception here - instead of relying on killTaskIfInterrupted to do the right thing (it is interrupted already according to the case check)
Not only will it not remove the unreachable `null`, but also ensure future changes to `killTaskIfInterrupted` or interrupt reset, etc does not break this.